### ISSUE TITLE
[gym] Confusing scheme selection if xcode project and workspace contain identically named scheme

### ISSUE THREAD

**Issue by janpio** [fastlane/fastlane#10943]
If you have an identically named scheme in both my xcode projcet and in my xcode workspace (Cordova projects are set up that way: project has Cordova, CordovaLib and MyApp, workspace has additional MyApp that is shared) this results in this confusing scheme selection output for `gym`:

```
...
[23:15:32]: -----------------
[23:15:32]: --- Step: gym ---
[23:15:32]: -----------------
[23:15:32]: $ xcodebuild -list -workspace platforms/ios/FastlaneExample.xcworkspace -configuration Debug
Select Scheme: 
1. CordovaLib
2. FastlaneExample
3. Cordova
2. FastlaneExample
?  2
[23:15:46]: $ xcodebuild -showBuildSettings -workspace platforms/ios/FastlaneExample.xcworkspace -scheme FastlaneExample -configuration Debug
```

Both "FastlaneExample" entries get number "2", I can't really decide which one to select. A solution could be to add "(Workspace)" and "(Project)" behind the selections if there are identically named selections.

But as the executed command below shows, it chooses the scheme name as a string as a parameter, so it can't really differentiate between the both anyway. Is it actually important at all which one is selected?  Then a solution could be to only show the selection entry once instead of twice.

**Comment by jdee** [comment]
Where does the workspace keep the scheme? In most cases I've seen, the workspace loads schemes from the individual projects. They are under `MyProject.xcodeproj/xcshareddata/xcschemes` for shared schemes or `MyProject.xcodeproj/xcuserdata/$USER.xcuserdata/xcschemes`. Is your workspace scheme stored somewhere under `MyProject.xcworkspace`?

**Comment by jdee** [comment]
What is the output of the `xcodebuild -list` command above? That's being parsed by `FastlaneCore::XcodebuildListOutputParser`, which is pretty simple. It probably just appears more than once, and maybe a `uniq` call somewhere is in order.

**Comment by janpio** [comment]
Good question.

Here is the answer:

```
MacBook-Pro:ios sujan$ tree -f | grep xcscheme
│   │           └── ./CordovaLib/CordovaLib.xcodeproj/xcuserdata/sujan.xcuserdatad/xcschemes
│   │               ├── ./CordovaLib/CordovaLib.xcodeproj/xcuserdata/sujan.xcuserdatad/xcschemes/Cordova.xcscheme
│   │               ├── ./CordovaLib/CordovaLib.xcodeproj/xcuserdata/sujan.xcuserdatad/xcschemes/CordovaLib.xcscheme
│   │               └── ./CordovaLib/CordovaLib.xcodeproj/xcuserdata/sujan.xcuserdatad/xcschemes/xcschememanagement.plist
│           └── ./FastlaneExample.xcodeproj/xcuserdata/sujan.xcuserdatad/xcschemes
│               ├── ./FastlaneExample.xcodeproj/xcuserdata/sujan.xcuserdatad/xcschemes/FastlaneExample.xcscheme
│               └── ./FastlaneExample.xcodeproj/xcuserdata/sujan.xcuserdatad/xcschemes/xcschememanagement.plist
│   │   └── ./FastlaneExample.xcworkspace/xcshareddata/xcschemes
│   │       └── ./FastlaneExample.xcworkspace/xcshareddata/xcschemes/FastlaneExample.xcscheme
```

So 2 from the xcodeproj of the Lib, 1 from my xcodeproj and 1 from my workspace (that is also the shared one).

**Comment by jdee** [comment]
I set up something kind of similar. I actually have no idea how to add a scheme to a workspace in Xcode. But I set up the same scheme in two different projects. In Xcode I see this:

<img width="318" alt="screen shot 2017-11-18 at 10 32 49 am" src="https://user-images.githubusercontent.com/121951/32983627-6f0c8232-cc4c-11e7-8198-a4198ba0197d.png">

But from xcodebuild:

```bash
[jdee@Jimmy-Dees-MacBookPro TestProject (tmp)]$ xcodebuild -list -workspace TestProject.xcworkspace -configuration Debug
Information about workspace "TestProject":
    Schemes:
        TestProject
        TestProject

```

All `gym` can do is pass one of those strings to `xcodebuild -scheme`, so I doubt there's a real way around it. But `xcodebuild -list` doesn't always show available schemes. I've been working around this elsewhere. Fastlane could also do something different here.
